File: linkedlist/singly_linked_list/basic_linked_list/linked_list.py
#Implementation of Linked List

class node:
    def __init__(self ,data):
        self.data = data
        self.next = None

class linked_list:
    def __init__(self):
        self.head = None

if __name__ == '__main__':
    link_list = linked_list()
    
    link_list.head = node(0)
    
    scnd = node(1)
    scnd.next = None

    link_list.head.next = scnd

    #traverse through linked-list
    while link_list.head != None:
        print(link_list.head.data ,end="\t")
        link_list.head = link_list.head.next

File: linkedlist/singly_linked_list/basic_linked_list/linked_list.c
#include <stdio.h>
#include <stdlib.h>

struct node{
    int data;
    struct node *next;
};

void print_linked_list(struct node *HEAD){
    while(HEAD != NULL){
        printf("%d\t" ,HEAD->data);
        HEAD = HEAD->next;
    }
}

int main(){
    struct node *HEAD;
    
    struct node *first = NULL;
    struct node *scnd  = NULL;

    first = malloc(sizeof(struct node));
    scnd  = malloc(sizeof(struct node));

    first->data = 1;

    scnd->data = 2;
    first->next = scnd;

    scnd->next = NULL;

    HEAD = first;
    
    print_linked_list(HEAD);
    return 0;
}